elephant/asset.py

```diff
@@ -1128,34 +1159,43 @@ def _jsf_uniform_orderstat_3d(u, alpha, n, verbose=False):
 
 def _pmat_neighbors(mat, filter_shape, n_largest):
     """
-    Build the 3D matrix L of largest neighbors of elements in a 2D matrix mat.
+    Build the 3D matrix `L` of largest neighbors of elements in a 2D matrix `mat`.
 
-    For each entry mat[i, j], collects the nr_largest elements with largest
-    values around mat[i,j], say z_i, i=1,2,...,nr_largest, and assigns them
-    to L[i, j, :].
-    The zone around mat[i, j] where largest neighbors are collected from is
-    a rectangular area (kernel) of shape (l, w) = filter_shape centered around
-    mat[i, j] and aligned along the diagonal.
-    If mat is symmetric, only the triangle below the diagonal is considered.
+    For each entry `mat[i, j]`, collects the `n_largest` elements with largest
+    values around `mat[i, j]`, say `z_i, i=1,2,...,n_largest`, and assigns them
+    to `L[i, j, :]`.
+    The zone around `mat[i, j]` where largest neighbors are collected from is
+    a rectangular area (kernel) of shape `(l, w) = filter_shape` centered around
+    `mat[i, j]` and aligned along the diagonal.
+    
+    If `mat` is symmetric, only the triangle below the diagonal is considered.
 
     Parameters
     ----------
     mat : np.ndarray
-        a square matrix of real-valued elements
-
-    filter_shape : tuple
-        a pair (l, w) of integers representing the kernel shape
-
-    n_largest : int, optional
-        the number of largest neighbors to collect for each entry in mat
-        If None (default) the filter length l is used
-        Default: 0
+        A square matrix of real-valued elements.
+    filter_shape : tuple of int
+        A pair of integers representing the kernel shape `(l, w)`.
+    n_largest : int
+        The number of largest neighbors to collect for each entry in `mat`.
+        If None, the filter length `l` in `filter_shape` is used.
 
     Returns
     -------
-    L : np.ndarray
-        a matrix of shape (nr_largest, l, w) containing along the first
-        dimension lmat[:, i, j] the largest neighbors of mat[i, j]
+    lmat : np.ndarray
+        A matrix of shape `(nr_largest, l, w)` containing along the first
+        dimension `lmat[:, i, j]` the largest neighbors of `mat[i, j]`.
+        
+    Raises
+    ------
+    AssertionError
```

Comment:
Could replace the `assert` to raise ValueError, since assertions were not supposed to be used in Elephant